### ISSUE TITLE
Please default to allow destination moz-extension://*:*

### ISSUE THREAD

**Issue by madduck** [RequestPolicyContinued/requestpolicy#835]
Without a rule allowing `→ moz-extension://*:*`, extensions such as Ghostery, PrivacyBadger and PassFF don't work anymore. I think it'd be safe to add this rule to the default ruleset.

See also https://github.com/passff/passff/issues/211

**Comment by myrdd** [comment]
Is `moz-extension` the destination scheme or the origin scheme? `moz-extension` as destination should work.

**Comment by myrdd** [comment]
Which version of RP do you use?

**Comment by myrdd** [comment]
> Which version of RP do you use?

If you're using 0.5.28.\*, use 0.5.31 instead:
https://addons.mozilla.org/en-US/firefox/addon/requestpolicy-legacy/

**Comment by madduck** [comment]
I'm using RP-Continued at the latest version from addons.mozilla.org, 1.0.beta12.4.

**Comment by madduck** [comment]
I'm using RP-Continued at the latest version from addons.mozilla.org, 1.0.beta12.4.

**Comment by myrdd** [comment]
Please provide a bug report like shown in [this template](https://github.com/RequestPolicyContinued/requestpolicy/blob/dev-1.0/.github/ISSUE_TEMPLATE.md).

**Comment by madduck** [comment]
Hm, if you think…

Browser and Add-ons:

Browser: Firefox 55.0
RequestPolicy version: 1.0.beta12.4
Other add-ons: PassFF 1.0
RequestPolicy settings:

Default policy: deny
Steps to reproduce:

1. Create a new browser profile
2. Install RPC, and then PassFF.
3. Notice how the PassFF addon's toolbar button has no icon, and nothing happens when you click on it.
4. Add a policy to allow requests to destination `moz-extension://*:*`
5. Disable the PassFF add-on, and re-enable it
6. Notice how now the toolbar button has an icon and a menu pops up when you click on it.

**Comment by myrdd** [comment]
For the record: passff 1.0 is available here: https://github.com/passff/passff/releases/tag/1.0.0

I'm sorry @madduck but I'm still unable to reproduce your problem. One problem I face is that I don't have the native application installed. These are the errors I get:

```
[PassFF] Error detecting GPG Agent: Error: Attempt to postMessage on disconnected port
Stack trace:
Async*init/PassFF.Preferences<@moz-extension://1dde8e3a-1968-40a1-9727-adaab877ab59/modules/preferences.js:93:18
promise callback*init@moz-extension://1dde8e3a-1968-40a1-9727-adaab877ab59/modules/preferences.js:67:14
init@moz-extension://1dde8e3a-1968-40a1-9727-adaab877ab59/modules/main.js:112:12
@moz-extension://1dde8e3a-1968-40a1-9727-adaab877ab59/background.js:4:1
  main.js:16:7

[PassFF] Error initializing preferences: Error: Attempt to postMessage on disconnected port
Stack trace:
Async*init_env@moz-extension://1dde8e3a-1968-40a1-9727-adaab877ab59/modules/pass.js:69:12
init@moz-extension://1dde8e3a-1968-40a1-9727-adaab877ab59/modules/pass.js:74:12
init/<@moz-extension://1dde8e3a-1968-40a1-9727-adaab877ab59/modules/main.js:115:18
promise callback*init@moz-extension://1dde8e3a-1968-40a1-9727-adaab877ab59/modules/main.js:112:12
@moz-extension://1dde8e3a-1968-40a1-9727-adaab877ab59/background.js:4:1
```

I still think RP already does what it should. the `moz-extension` scheme is already whitlisted [[source code](https://github.com/RequestPolicyContinued/requestpolicy/blob/v1.0.beta12.4/src/content/lib/request.jsm#L155)] in RPC `1.0.beta12.4`.

Since this add-on requires a native application to be installed, you could help me by providing a video how you reproduce the problem in a new browser profile. In any case, check the Browser Console (ctrl+shift+J) for errors. If you don't want to post a video link publicly, send it to me via email.

**Comment by madduck** [comment]
I found that just allowing `moz-extension://` doesn't work, I had to 
explicitly enter a `*` for host *and* port. Could this make a difference?

I can try to produce a video, but I'm actually going to be AFK for a few days before I can try that. Maybe we can find another solution. For me, both PrivacyBadger and Ghostery are also affected by the same problem.

**Comment by myrdd** [comment]
> I found that just allowing `moz-extension://` doesn't work, I had to 
> explicitly enter a `*` for host *and* port. Could this make a difference?

For user rules it makes a difference, but not for the add-on internal whitelist.

> I can try to produce a video, but I'm actually going to be AFK for a few days before I can try that.

No worries, you can do it when you're back, or we see how else track down the issue.

> Maybe we can find another solution. For me, both PrivacyBadger and Ghostery are also affected by the same problem.

Let's try with Ghostery. I've tested RPC and Ghostery together just now, and I can see that Ghostery's request counter (on the ghostery icon) is zero on many sites. Other than that, I did not discover any issues, nor any changes of Ghostery's behavior when allowing requests to `moz-extension://*:*` like you mentioned.—Please again provide a step-by-step guide for your Ghostery problem, just like you did for PassFF.

**Comment by myrdd** [comment]
This issue has been fixed in commit fcc9d088d08f0a091ae8cc316807c181299d4ad3, released in v1.0.beta13.0.
